CMakeLists.txt

```diff
@@ -63,7 +63,8 @@ if(BUILD_STATIC)
         set_target_properties(xmp_static PROPERTIES OUTPUT_NAME xmp)
     endif()
 
-    target_compile_definitions(xmp_static PRIVATE ${LIBXMP_DEFINES} -DBUILDING_STATIC)
+    target_compile_definitions(xmp_static PRIVATE ${LIBXMP_DEFINES})
+    target_compile_definitions(xmp_static PUBLIC -DBUILDING_STATIC)
```

Comment:
I don't like this being public - `BUILDING_STATIC` is unfortunately a fairly common macro name

Reply:
It's here is a public because of the #427

Reply:
> It's here is a public because of the #427, unit tests and examples built in the same CMake build will fail to link on Windows

IMHO, the library user should know whether he is using the static build or not, and define `BUILDING_STATIC` accordingly. Others may think differently.

Reply:
Right now, this is the target of a static library, and so, to link it, the `BUILDING_STATIC` macro will be needed on Windows. I'll make it being private again, however, I ask, please remove the `dllimport()` at all, it adds extra pain with the necessity to define the `BUILDING_STATIC` macro at the user application side to simply get static libxmp being linked. If this macro is a common thing, it may affect the user application that wants to link the libxmp statically. That is why I ague to remove `dllimport()` at all.

Reply:
@cmatsuoka (and possibly @AliceLR) should decide. (mentioned their names in #427)

Reply:
Okay, I made the -DBUILDING_STATIC being private, and so, the real solution should be done at #427